growing bucket optimization for greedy text generation (#417)
huggingface/optimum-habana: examples/text-generation/README.md

--prompt "Here is my prompt" \
--peft_model trl-lib/llama-7b-se-rm-peft
```

### Using growing bucket optimization

> With `--bucketsize`, instead of padding up the kvcache up to full size before starting, we grow the cache/input in multiples of bucketsize. This helps increase throughput and also reduce number of compilations if teh dataset has varying prompt lengths.

Reviewer: ```suggestion
With `--bucket_size`, instead of padding up the kv-cache up to full size before starting, we grow the cache/input in multiples of `bucket_size`. This helps increase throughput and also reduce number of compilations if the dataset has varying prompt lengths.
```

> For now, it is available only for greedy generation, and cannot be used with `--reuse_cache`

Reviewer: ```suggestion
> For now, it is available only for greedy generation, and cannot be used with `--reuse_cache`.
```

> ```

Reviewer: ```suggestion

Here is an example:
```
```

python run_generation.py \

> --model_name_or_path path_to_llama_or_falcon  \

Reviewer: ```suggestion
--model_name_or_path path_to_model  \
```
It should be compatible with any model right?

--use_hpu_graphs \
--use_kv_cache \
--bf16 \
--max_new_tokens 200 \
--batch_size=2 \

> --bucketsize 50

Reviewer: ```suggestion
--bucket_size 50
```
